src/lib/constants/sectionListView/viewConfigResolver.ts

```diff
@@ -0,0 +1,142 @@
+import { uniqueBy } from '../../utils'
+import { getTranslatedProperty } from '../translatedModelProperties'
+import {
+    defaultModelViewConfig,
+    modelListViewsConfig,
+    ViewConfigPart,
+    ModelPropertyConfig,
+    ModelPropertyDescriptor,
+    FilterDescriptor,
+    FilterConfig,
+} from './sectionListViewsConfig'
+
+interface ResolvedViewConfigPart<TEntry> {
+    available: ReadonlyArray<TEntry>
+    default: ReadonlyArray<TEntry>
+}
+interface ResolvedViewConfig {
+    columns: ResolvedViewConfigPart<ModelPropertyDescriptor>
+    filters: ResolvedViewConfigPart<FilterDescriptor>
+}
+
+interface ResolvedSectionListView {
+    [key: string]: ResolvedViewConfig
+}
+
+const toModelPropertyDescriptor = (
+    propertyConfig: ModelPropertyConfig
+): ModelPropertyDescriptor => {
+    if (typeof propertyConfig === 'string') {
+        return {
+            label: getTranslatedProperty(propertyConfig),
+            path: propertyConfig,
+        }
+    }
+    return propertyConfig
+}
+
+const toFilterDescriptor = (propertyConfig: FilterConfig): FilterDescriptor => {
+    if (typeof propertyConfig === 'string') {
+        return {
+            label: getTranslatedProperty(propertyConfig),
+            filterKey: propertyConfig,
+        }
+    }
+    return propertyConfig
+}
+
+const resolveFilterConfig = (
+    part: ViewConfigPart<FilterConfig>
+): ResolvedViewConfigPart<FilterDescriptor> => {
+    const { default: defaultFilters, available: defaultAvailableFilers } =
+        defaultModelViewConfig.filters
+
+    const mergedAvailableDescriptors = uniqueBy(
+        [
+            part.available || [],
+            part.overrideDefaultAvailable ? [] : defaultAvailableFilers || [],
+            part.default || [],
+        ]
+            .flat()
```

Comment:
```suggestion
        [
            ...(part.available || []),
            ...(part.overrideDefaultAvailable ? [] : defaultAvailableFilers || []),
            ...(part.default || []),
        ]
```

Reply:
This seems very personal preference. I'm not a big fan of `...( condition ? true: false)` when it can be avoided (can be useful for conditionally adding keys to an object). To me `.flat` reads cleaner here.

Reply:
Hence the wording "Suggested change" :) Leave it as is